plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/libraryElement/impl/AttributeAnnotations.java

```diff
@@ -30,6 +40,34 @@ public static String getQualifiedName(final Attribute attribute) {
 		return NamedElementAnnotations.getQualifiedName(attribute);
 	}
 
+	public static boolean validateAttributeDeclaration(final Attribute attribute, final DiagnosticChain diagnostics,
+			final Map<Object, Object> context) {
+		if (null != attribute.getAttributeDeclaration()
+				&& attribute.getAttributeDeclaration().getTypeEntry() instanceof ErrorTypeEntry) {
+			if (diagnostics != null) {
+				diagnostics.add(createAttributeValidationDiagnostic(MessageFormat.format(
+						Messages.ErrorMarkerInterfaceAnnotations_MissingAttribute, attribute.getName()), attribute));
+			}
+			return false;
+		}
+		return true;
+	}
+
+	private static Diagnostic createAttributeValidationDiagnostic(final String message, final Attribute attribute) {
+		return new BasicDiagnostic(Diagnostic.ERROR, LibraryElementValidator.DIAGNOSTIC_SOURCE,
+				LibraryElementValidator.ERROR_MARKER_INTERFACE__VALIDATE_ATTRIBUTES, message,
+				FordiacMarkerHelper.getDiagnosticData(attribute, getAttributeFeature(attribute), attribute.getName()));
```

Comment:
Please use the correct code and you can also reference the feature directly here:
```suggestion
				LibraryElementValidator.ATTRIBUTE__VALIDATE_ATTRIBUTE_DECLARATION, message,
				FordiacMarkerHelper.getDiagnosticData(attribute, LibraryElementPackage.Literals.ATTRIBUTE__ATTRIBUTE_DECLARATION, attribute.getName()));
```

Reply:
I then also changed the LibraryElementValidator in the FordiacMarkerResolutionGenerator, to get the Quickfix to work